lib/Net/SAML2/IdP.pm

```diff
@@ -225,14 +228,18 @@ around BUILDARGS => sub {
         for my $use (keys %{$params{certs}}) {
             my $certs = $params{certs}{$use};
             for my $pem (@{$certs}) {
+                my @errors;
                 my $cert = Crypt::OpenSSL::X509->new_from_string($pem);
-                ## BUGBUG this is failing for valid things ...
-                eval { $ca->verify($cert) };
-                if ($@) {
-                    warn "Can't verify IdP cert: $@";
-                    next;
+                try {
+                    $ca->verify($cert);
+                    push(@{$certificates{$use}}, $pem);
                 }
-                push(@{$certificates{$use}}, $pem);
+                catch {push (@errors, $_);}
```

Comment:
I think the catch { } finally block can be one and the same if it stays in the loop. Because we can just run the finally block in the catch. Having the `if ($debug && @errors)` should be outside of the loop and we don't need the `finally` here.